File: apps/web/lib/api/registration.ts
export type RegisterPayload = {
  full_name: string;
  prn: string;
  email: string;
  department: string;
  year_of_study: string;
};

export type CompanyOut = {
  name: string;
  sector: string;
  backstory: string;
  unique_strength: string;
  unique_weakness: string;
  unique_passive_ability: string;
  cash: number;
  employees: number;
};

export type RegisterResponse = {
  player_id: string;
  session_id: string;
  resumed: boolean;
  company: CompanyOut;
};

const API_BASE = process.env.NEXT_PUBLIC_API_BASE_URL ?? "http://localhost:8000";

export async function registerPlayer(payload: RegisterPayload): Promise<RegisterResponse> {
  const res = await fetch(`${API_BASE}/api/v1/players/register`, {
    method: "POST",
    credentials: "include",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    const body = await res.json().catch(() => null);
    throw new Error(body?.detail ?? "Registration failed.");
  }

  return res.json();
}

File: apps/api/app/schemas/registration.py
import uuid

from pydantic import BaseModel, EmailStr, Field


class RegisterRequest(BaseModel):
    full_name: str = Field(min_length=1)
    prn: str = Field(min_length=1)
    email: EmailStr
    department: str = Field(min_length=1)
    year_of_study: str = Field(min_length=1)


class CompanyOut(BaseModel):
    name: str
    sector: str
    backstory: str
    unique_strength: str
    unique_weakness: str
    unique_passive_ability: str
    cash: float
    employees: int


class RegisterResponse(BaseModel):
    player_id: uuid.UUID
    session_id: uuid.UUID
    company_id: uuid.UUID
    current_quarter: int
    resumed: bool
    company: CompanyOut